src/utils/dex/location.js

```diff
@@ -0,0 +1,147 @@
+const { displayNames, areaNames,  mapInfo, field_items, hidden_items } = require('../../../__gamedata');
+const { areasList } = require('../../../__gamedata/areas');
+
+// This first section is using the areas_updated.csv for its data
+const AreaMap = {}
+const lines = areasList.split('\n');
+const headers = lines[0].split(",");
+lines.slice(1).forEach(line => {
+  const values = line.split(",");
+  const zoneId = parseInt(values[0]);
+  AreaMap[zoneId] = Object.fromEntries(headers.map((header, index) => [header, values[index]]));
+});
+
+function getZoneCodeFromCSV(zoneId) {
+  if (!zoneId) {
+    return null;
+  }
+  return AreaMap[zoneId].Zone_Code;
```

Comment:
Consider adding a safe check to prevent errors if zoneId is not in AreaMap
```suggestion
  return AreaMap[zoneId]?.Zone_Code;
```